src/Microsoft.SqlTools.ServiceLayer/QueryExecution/Query.cs

```diff
@@ -489,6 +494,15 @@ private async Task ExecuteInternal()
                 // Call the query execution callback
                 if (QueryCompleted != null)
                 {
+                    if (sqlConn != null)
+                    {
+                        // Update SPID here as it may change upon restart and only updates upon query execution.
+                        SPID = (sqlConn.GetUnderlyingConnection() as SqlConnection).ServerProcessId;
+                        if (SPID != 0)
+                        {
+                            Console.WriteLine(SPID);
```

Comment:
Console.WriteLine is going to break STS (stdout is used for the JSONRPC channel so this would be inserting random characters in that stream)